tests/test_tasks.py

```diff
@@ -1,16 +1,117 @@
 # -*- coding: utf-8 -*-
 #
 # Copyright (C) 2022-2024 CERN.
+# Copyright (C) 2025 Northwestern University.
 #
 # Invenio App RDM is free software; you can redistribute it and/or modify it
 # under the terms of the MIT License; see LICENSE file for more details.
 """Test invenio-app-rdm celery tasks."""
 
-from invenio_app_rdm.tasks import file_integrity_report
+from io import BytesIO
+
+import pytest
+from invenio_access.permissions import system_identity
+from invenio_db import db
+from invenio_files_rest.models import Bucket, FileInstance, Location, ObjectVersion
+from invenio_rdm_records.proxies import current_rdm_records
 
+from invenio_app_rdm.tasks import file_integrity_report
 
-def test_task_file_integrity_report(app, invalid_file_instance):
+# Fixtures
+
+
+@pytest.fixture
+def draft_with_file_instance(running_app, minimal_record):
+    """Create draft with file and return (Draft, FileInstance)."""
+    minimal_record["files"] = {"enabled": True}
+
+    record_service = current_rdm_records.records_service
+    file_service = record_service.draft_files
+
+    draft = record_service.create(system_identity, minimal_record)
+    file_to_initialise = [
+        {
+            "key": "article.txt",
+            "checksum": "md5:c785060c866796cc2a1708c997154c8e",
+            "size": 17,  # 2kB
+            "metadata": {
+                "description": "Published article PDF.",
+            },
+        }
+    ]
+    file_service.init_files(system_identity, draft.id, file_to_initialise)
+    content = BytesIO(b"test file content")
+    file_service.set_file_content(
+        system_identity,
+        draft.id,
+        file_to_initialise[0]["key"],
+        content,
+        content.getbuffer().nbytes,
+    )
+    file_item = file_service.commit_file(system_identity, draft.id, "article.txt")
+    # fmt: off
+    id_of_file_instance = (
+        file_item
+        ._record
+        .files
+        .entries["article.txt"]
+        .object_version.file_id
+    )
+    # fmt: on
+    f = FileInstance.query.get(id_of_file_instance)
+    return draft, f
+
+
+@pytest.fixture
+def draft_with_invalid_file_instance(draft_with_file_instance):
+    # Force an invalid checksum
+    draft, f = draft_with_file_instance
+    f.checksum = "invalid"
+    f.verify_checksum()
+    db.session.commit()
+
+    # Retrieve the file instance (with updated last_check)
+    f = FileInstance.query.get(f.id)
+
+    return draft, f
+
+
+@pytest.fixture
+def record_with_invalid_file_instance(draft_with_invalid_file_instance):
+    draft, f = draft_with_invalid_file_instance
+    record_service = current_rdm_records.records_service
+    return record_service.publish(system_identity, draft.id), f
+
+
+@pytest.fixture
+def invalid_file_instance(record_with_invalid_file_instance):
+    draft, f = record_with_invalid_file_instance
+    return f
+
+
+@pytest.fixture
+def set_tmp_config(app):
```

Comment:
Actually made a PR to pytest-invenio to add this because leaking tests caused so much headache: https://github.com/inveniosoftware/pytest-invenio/pull/117